std/assembly/typedarray.ts

```diff
@@ -1718,3 +1762,41 @@ function WRAP<TArray extends ArrayBufferView, T>(buffer: ArrayBuffer, byteOffset
   store<usize>(out, changetype<usize>(buffer) + <usize>byteOffset, offsetof<TArray>("dataStart"));
   return changetype<TArray>(out); // retains
 }
+
+// @ts-ignore: decorator
+@inline
+function SET<TArray extends ArrayBufferView, T, UArray extends ArrayBufferView, U>(target: TArray, source: UArray, offset: i32 = 0): void {
+  // need to assert at compile time that U is not a reference or a function
+  if (isReference<U>()) {
+    ERROR(E_NOTIMPLEMENTED);
+  }
+
+  // Uncaught RangeError: offset is out of bounds
+  if (offset < 0) throw new RangeError(E_INDEXOUTOFRANGE);
+  if (source.length + offset > target.length) throw new RangeError(E_INDEXOUTOFRANGE);
+
+  let targetOffset = <usize>offset << alignof<T>();
+  let targetDataStart = target.dataStart + targetOffset;
+  let sourceDataStart = source.dataStart;
+
+  // if the types align and match, use memory.copy() instead of manual loop
+  if (isInteger<T>() == isInteger<U>() && alignof<T>() == alignof<U>()) {
+    memory.copy(targetDataStart, sourceDataStart, source.byteLength);
+  } else {
+    let count = source.length;
+    for (let i = 0; i < count; i++) {
+      let targetStoreOffset = <usize>i << alignof<T>();
+      let sourceLoadOffset = <usize>i << alignof<U>();
```

Comment:
`targetStoreOffset` and `sourceLoadOffset` better move inplace (inlined) inside `store`

Reply:
Won't this just get optimized by binaryen?

Reply:
I guess not necessarily.

Reply:
The `if` statement is inlined because the conditions are statically known at runtime.

Reply:
I mean use `load(ptr + (i << alignof<T>()))` instead  `load(ptr + cachedStoreOffset)`. That's that case when caching is not helps

Reply:
Oh that makes a lot of sense. No problem!